src/software/multithreading/observer_test.cpp

```diff
@@ -55,7 +55,18 @@ TEST(Observer, getDataReceivedPerSecond_time_buffer_filled)
         test_observer.receiveValue(i);
         std::this_thread::sleep_for(std::chrono::milliseconds(10));
     }
-    EXPECT_NEAR(test_observer.getDataReceivedPerSecond(), 5 / (4 * 0.01), 10);
+    EXPECT_NEAR(test_observer.getDataReceivedPerSecond(), 1 / 0.01, 20);
+}
+
+TEST(Observer, getDataReceivedPerSecond_time_buffer_filled_twice_over)
+{
+    TestObserver test_observer;
+    for (unsigned int i = 0; i < TestObserver::TIME_BUFFER_SIZE * 2; i++)
+    {
+        test_observer.receiveValue(i);
+        std::this_thread::sleep_for(std::chrono::milliseconds(5));
+    }
+    EXPECT_NEAR(test_observer.getDataReceivedPerSecond(), 1 / 0.005, 20);
```

Comment:
Hmm I think the order of receiveValue and threadsleep should be swapped? If we're using 1/interval, that means we (1/interval) values/second, so at t=0 we would receive 0 values, but we actually get one since we receive before we sleep? Not too sure about this though.

Reply:
so the receive_time_buffer pushes back the current time, i.e. `std::chrono::steady_clock::now()` every time it receives a value, so sleeping first won't do anything. The observer won't know that the thread slept before the first message, it only knows/calculates the time between messages received

Reply:
Oh I see, thanks for the explanation!